### PR TITLE
Fixing verify ceritificate options on https listeners

### PR DESCRIPTION
**Pull request by JanuszJakubiec**
The `verify_mode` option had no effect on incoming HTTP connections, and always works as `none`.
In this PR, I've enabled the check and disallowed  connections from untrusted peers with the `peer` mode. 
The option `selfsigned_peer` is supported as well.

The changes in the code consist of adding the option `fail_if_no_peer_cert = true`, whenever the verify_mode is set to `peer` or `selfsigned_peer`. Because the option is desired only in https listeners, not when we are connecting to the database, a separate function was created only for cowboy listener, to allow it to work properly.

In the big_tests the tests are testing all possible scenarios, with graphql handlers: `domain_admin_listener` has verify_mode set to `none`, `user_listener` has verify_mode set to `selfsigned_peer`, and `admin_listener` has veirfy_mode set to `peer`. Certificates needed to perform the tests were signed based on the existing code in `sasl_external_suite`. To prevent code duplication, the new helper, `ca_certificate_helper` has been created.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/210512) / elasticsearch_and_cassandra_mnesia / 59648f750194d6c2246d227eeb80c5af98d16656
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4236/210512/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4236/210512/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%4008b983a6f136.2024-03-06_09.19.10/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/210545) / elasticsearch_and_cassandra_mnesia / 650c8cfb93272cb2b4fc16d13e0a05e5d6e9d3b5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4236/210545/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4236/210545/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40f2172993b90c.2024-03-06_09.28.22/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/210649) / elasticsearch_and_cassandra_mnesia / 6f2fc70ea4f9470c370585dae4bd6604ccce5bed
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4236/210649/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4236/210649/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%406fcb4c699777.2024-03-06_10.03.35/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/210673) / elasticsearch_and_cassandra_mnesia / df890bc4e6ea2ce9cb7b2bd2a30e5d625cac99bf
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4236/210673/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4236/210673/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%404d02cc49893b.2024-03-06_10.07.13/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/211050) / elasticsearch_and_cassandra_mnesia / 6be7582a02a78e873656e0486d2764559b73862a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4236/211050/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4236/211050/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%404f8bed0a2f2a.2024-03-08_08.59.46/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Review by pawlooss1 — COMMENTED** [review]
I left some comments. Please check them out.

File: big_tests/tests/graphql_SUITE.erl [hunk 1]
```diff
@@ -185,8 +206,119 @@ multiple_categories_query_test(Config) ->
...
+    ?assertMatch({ok, {{<<"400">>, <<"Bad Request">>}, _, _, _, _}}, Result).
+
 %% Helpers
 
+assert_match_result({error, {tls_alert, {certificate_required, _}}}) ->
+    ok;
+assert_match_result({error, connection_closed}) ->
+    ok;
+assert_match_result(_) ->
+    wrong_pattern.
```
[the remark this answers is not in the document]
Comment (JanuszJakubiec) [reply big_tests/tests/graphql_SUITE.erl:277]: The error that is returned should be: `{error, {tls_alert, {certificate_required, _}}}`, however sometimes, the tests results in {error, connection_closed}. I've decided It's not the subject of this task to debug and check the code responsible for testing the connection. I believe this test is important, to show that connection without certificate is not working properly.

File: big_tests/tests/graphql_SUITE_data/sign_cert.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+req_file=
+out_file=
+
+## process input parameters
+[ $# -eq 0 ] && invalid_usage
+while true; do
+  [ "$1" == "--req" ] && [ $# -gt 1 ] && { req_file="$2";  shift 2; continue; }
+  [ "$1" == "--out" ] && [ $# -gt 1 ] && { out_file="$2";  shift 2; continue; }
+  break
+done
```
[the remark this answers is not in the document]
Comment (JanuszJakubiec) [reply big_tests/tests/graphql_SUITE_data/sign_cert.sh:19]: I believe the existing code is better than:
`  if [ "$1" == "--req" ] && [ $# -gt 1 ]; then
    req_file="$2"
    shift 2
  elif [ "$1" == "--out" ] && [ $# -gt 1 ]; then
    out_file="$2"
    shift 2
  else
    break
  fi`

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/211299) / elasticsearch_and_cassandra_mnesia / 2e254a02c4f6e0b2e39f64e7e761c22c55231eba
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4236/211299/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4236/211299/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%405faefcf32761.2024-03-11_08.55.42/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

File: big_tests/tests/graphql_SUITE_data/sign_cert.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (pawlooss1) [reply big_tests/tests/graphql_SUITE_data/sign_cert.sh:19]: You're right. Let's leave it this way. I was just initially scared when I saw `continue` 😅

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/212276) / elasticsearch_and_cassandra_mnesia / 2d8c9c62dc0640a3aa30639eb517832b027a3d37
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4236/212276/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4236/212276/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%404d08e72824f0.2024-03-14_09.22.39/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Review by pawlooss1 — APPROVED** [review]
❤️ 👍

File: big_tests/tests/graphql_SUITE.erl (hunk 1 above)
[the remark this answers is not in the document]
Comment (pawlooss1) [reply big_tests/tests/graphql_SUITE.erl:278]: After discussion we agreed to left a comment about it in the code.

**Pull request merged**